src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -1501,6 +1519,83 @@ public async Task<SubscriptionInfo> GetSubscriptionAsync(ISubscriber subscriber)
             return subscriptionInfo;
         }
 
+        public async Task<TaxInfo> GetTaxInfoAsync(ISubscriber subscriber)
+        {
+            if (subscriber != null && !string.IsNullOrWhiteSpace(subscriber.GatewayCustomerId))
+            {
+                var customerService = new CustomerService();
+                var customer = await customerService.GetAsync(subscriber.GatewayCustomerId);
+
+                if (customer == null)
+                {
+                    return null;
+                }
+
+                var address = customer.Address;
+                var taxId = customer.TaxIds?.FirstOrDefault();
+
+                // Line1 is required, so if missing we're using the subscriber name
+                // see: https://stripe.com/docs/api/customers/create#create_customer-address-line1
+                if (address != null && address.Line1 == subscriber.BillingName())
+                {
+                    address.Line1 = null;
+                }
+
+                return new TaxInfo
+                {
+                    TaxIdNumber = taxId?.Value,
+                    BillingAddressLine1 = address?.Line1,
+                    BillingAddressLine2 = address?.Line2,
+                    BillingAddressCity = address?.City,
+                    BillingAddressState = address?.State,
+                    BillingAddressPostalCode = address?.PostalCode,
+                    BillingAddressCountry = address?.Country,
+                };
+            }
+
+            return null;
+        }
+
+        public async Task SaveTaxInfoAsync(ISubscriber subscriber, TaxInfo taxInfo)
+        {
+            if (subscriber != null && !string.IsNullOrWhiteSpace(subscriber.GatewayCustomerId))
+            {
+                var customerService = new CustomerService();
+                var customer = await customerService.UpdateAsync(subscriber.GatewayCustomerId, new CustomerUpdateOptions
+                {
+                    Address = new AddressOptions
+                    {
+                        Line1 = taxInfo.BillingAddressLine1 ?? subscriber.BillingName(),
+                        Line2 = taxInfo.BillingAddressLine2,
+                        City = taxInfo.BillingAddressCity,
+                        State = taxInfo.BillingAddressState,
+                        PostalCode = taxInfo.BillingAddressPostalCode,
+                        Country = taxInfo.BillingAddressCountry,
+                    },
+                });
+
+                if (!subscriber.IsUser() && customer != null)
+                {
+                    var taxIdService = new TaxIdService();
+                    var taxId = customer.TaxIds?.FirstOrDefault();
+
+                    if (taxId != null)
+                    {
+                        await taxIdService.DeleteAsync(customer.Id, taxId.Id);
+                    }
```

Comment:
I guess there is no "update"?

Reply:
Nope, even though a taxID has its own ID, they don't provide an update, only create and delete.